source/async_stack.cpp

```diff
@@ -99,6 +99,11 @@ static std::uint64_t get_os_thread_id() {
   return tid;
 #  elif defined(_WIN32)
   return std::uint64_t(GetCurrentThreadId());
+#  elif defined(__FreeBSD__)
+  long tid;
+  thr_self(&tid);
```

Comment:
Can't decide if it's worth adding BSD support here; you're not returning `tid` so this code is broken but we won't catch that since we don't have any known usage on BSD. Probably best to delete this.
```suggestion
```